File: backend/src/main/java/com/astral/express/pccms/medicine/entity/Medicine.java
package com.astral.express.pccms.medicine.entity;

import com.astral.express.pccms.common.domain.AuditableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.UUID;

@Entity
@Table(
    name = "medicines",
    uniqueConstraints = {
        @UniqueConstraint(columnNames = {"name", "unit"})
    }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Medicine extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "medicine_code", nullable = false, unique = true, length = 60)
    private String medicineCode;

    @Column(name = "name", nullable = false, length = 160)
    private String name;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private MedicineCategory category;

    @Column(name = "unit", nullable = false, length = 40)
    private String unit;

    @Column(name = "default_instruction", columnDefinition = "TEXT")
    private String defaultInstruction;

    @Column(name = "current_stock", nullable = false)
    @Builder.Default
    private Integer currentStock = 0;

    @Column(name = "unit_price_vnd", nullable = false, precision = 14, scale = 2)
    @Builder.Default
    private BigDecimal unitPriceVnd = BigDecimal.ZERO;

    @Column(name = "is_active", nullable = false)
    @Builder.Default
    private Boolean isActive = true;
}

File: frontend/src/types/medicine.ts
export interface MedicineResponse {
    id: string;
    medicineCode: string;
    name: string;
    categoryId: string;
    categoryName: string;
    unit: string;
    defaultInstruction: string;
    currentStock: number;
    unitPriceVnd: number;
    isActive: boolean;
}

export interface CreateMedicineRequest {
    name: string;
    categoryId: string;
    unit: string;
    defaultInstruction?: string;
    initialStock: number;
    unitPriceVnd: number;
}

export interface UpdateMedicineRequest {
    name?: string;
    categoryId?: string;
    unit?: string;
    defaultInstruction?: string;
    unitPriceVnd?: number;
}

export interface AddStockRequest {
    quantity: number;
    note?: string;
}
